packages/flutter_tools/lib/src/build_system/targets/native_assets.dart

```diff
@@ -197,6 +199,26 @@ class NativeAssets extends Target {
         case TargetPlatform.android_x64:
         case TargetPlatform.android_x86:
         case TargetPlatform.android:
+          final String? isAddToAppEnvironment =
```

Comment:
I am not a fan of this logic living in a large switch statement. Please pull out into a method and consider doing the same for the other platforms.